SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -37,12 +38,19 @@ class ApplicationPresenter : ApplicationContract.Presenter() {
     @OptIn(UnstableDefault::class)
 
     override fun onButtonPressed(origin: String, destination: String) {
-        val originCode = requireNotNull(codeMap[origin])
-        val destinationCode = requireNotNull(codeMap[destination])
-
-        launch {
-            val response = sequentialRequests(originCode, destinationCode)
-            view?.showData(response)
+        val originCode = codeMap[origin]
+        val destinationCode = codeMap[destination]
+        if (originCode != null && destinationCode != null) {
+            if (originCode != destinationCode) {
+                launch {
+                    val response = sequentialRequests(originCode, destinationCode)
+                    if (response != null) view?.showData(response)
+                }
+            } else {
+                view?.showAlert("Stations must be different")
+            }
+        } else {
+            view?.showAlert("Error: Station not found")
         }
```

Comment:
That's a lot of nesting. I wonder if we could, via magic of early returns, reduce it down a bit
For example, if both elements are null, you can show your alert and return, and it removes the need for else statement.